src/common/Requirements.ts

```diff
@@ -103,60 +104,88 @@ export function WrappedPromise(
         });
 }
 
-export abstract class BaseSetup implements RequirementList {
-    public skipBaseRequirements: boolean = false;
-    public skipAdditionalRequirements: boolean = false;
-    public requirements: Requirement[];
-    public additionalRequirements: Requirement[];
+export class CommandRequirement {
+    public baseRequirements: RequirementList = {
+        requirements: [],
+        enabled: true
+    };
+    public commandRequirements: RequirementList = {
```

Comment:
The separation of `baseRequirements` and `commandRequirements` is unclear to me. How will this differentiation work in practice? I had assumed we were going with a model of distinct requirements lists, and this would seem to indicate otherwise.

Reply:
I may have misinterpreted what was on the spec. This is pseudo-code from the spec:
![image](https://user-images.githubusercontent.com/42476999/113048520-7bcac280-9157-11eb-8333-b3b9e0e2ad71.png)
`commandRequirements` is what I wanted to emulate like `createRequirements` in the screenshot. My thinking was you probably wanted others like `hostRequirements` or `proxyRequirements`. Instead of specific names I am using the generic `commandRequirements`. 
So a command is composed with 2 requirements, `baseRequirements` which it can be used to compose prerequisite checks and `commandRequirements` which can focus on the command's specific requirement checks.